### PR TITLE
feat(build): Make freezing requirements idempotent

### PR DESCRIPTION
**Pull request by armenzg**
Running `make freeze-requirements` without changes to the base requirements file should not update to newer packages.

This makes it harder to make unintended changes

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -1,102 +1,102 @@
-beautifulsoup4>=4.7.1
```
Comment (armenzg) [inline requirements-base.txt:1]: Most changes in this file are just `s/>=/==`.

File: requirements-base.txt [hunk 2]
```diff
@@ -1,102 +1,102 @@
...
+google-cloud-pubsub==2.2.0
+google-cloud-storage==1.35.0
+google-cloud-functions==1.8.0
+google-cloud-spanner==3.17.0
+google-crc32c==1.3.0
+isodate==0.6.1
+jsonschema==3.2.0
+lxml==4.6.5
+maxminddb==2.0.3
+mistune==2.0.4
```
Comment (armenzg) [inline requirements-base.txt:29]: Using the same version as the one in the frozen file.

File: requirements-base.txt [hunk 3]
```diff
@@ -1,102 +1,102 @@
...
+croniter==0.3.37
+datadog==0.29.3
+django-crispy-forms==1.14.0
+django-pg-zero-downtime-migrations==0.11
+Django==2.2.28
+djangorestframework==3.12.4
+drf-spectacular==0.22.1
+email-reply-parser==0.5.12
+google-api-core==1.32.0
+google-auth==1.35.0
```
Comment (armenzg) [inline requirements-base.txt:16]: Using the same version as the one in the frozen file.

File: requirements-base.txt [hunk 4]
```diff
@@ -1,102 +1,102 @@
...
+snuba-sdk==1.0.1
+simplejson==3.17.6
+statsd==3.3
+structlog==21.1.0
+symbolic==8.7.1
+toronado==0.1.0
+typing-extensions==3.10.0.2
+ua-parser==0.10.0
+unidiff==0.7.4
+urllib3[brotli]==1.26.11
```
Comment (armenzg) [inline requirements-base.txt:67]: Using the same version as the one in the frozen file.

File: requirements-base.txt [hunk 5]
```diff
@@ -1,102 +1,102 @@
...
 # See if we can remove LDFLAGS from lib.sh
 # https://github.com/getsentry/sentry/pull/30094
 uWSGI==2.0.20.0
-zstandard>=0.18.0
+zstandard==0.18.0
 
-msgpack>=1.0.4
-cryptography>=3.4.8
+msgpack==1.0.4
+cryptography==37.0.2
```
Comment (armenzg) [inline requirements-base.txt:75]: Using the same version as the one in the frozen file.

File: requirements-base.txt [hunk 6]
```diff
@@ -1,102 +1,102 @@
...
-grpcio>=1.47.0
+grpcio==1.47.0
 
 # not directly used, but provides a speedup for redis
-hiredis>=0.3.1
+hiredis==0.3.1
 
 # not directly used, but pinned for at least semaphore/symbolic
-cffi>=1.15.0
+cffi==1.15.1
```
Comment (armenzg) [inline requirements-base.txt:90]: Using the same version as the one in the frozen file.

File: requirements-base.txt [hunk 7]
```diff
@@ -1,102 +1,102 @@
...
 # sentry-plugins specific dependencies
-phabricator>=0.7.0
+phabricator==0.7.0
 
 # test dependencies, but unable to move to requirements-test until
 # sentry.utils.pytest and sentry.testutils are moved to tests/
-selenium>=4.1.5
-sqlparse>=0.2.4,<=0.3.0
+selenium==4.3.0
+sqlparse==0.3.0
```
Comment (armenzg) [inline requirements-base.txt:102]: Using the same version as the one in the frozen file.

**Comment by armenzg** [comment]
Using a split view makes reviewing this easier:
image

**Review by asottile-sentry — CHANGES_REQUESTED** [review]

File: requirements-base.txt [hunk 8]
```diff
@@ -1,102 +1,102 @@
...
-python-u2flib-server>=5.0.0
-fido2>=0.9.2
-python3-saml>=1.14.0
-PyYAML>=5.4
-rb>=1.9.0
-redis-py-cluster>=2.1.0
-redis>=3.4.1
-requests-oauthlib>=1.2.0
-requests>=2.25.1
+beautifulsoup4==4.7.1
```
[the remark this answers is not in the document]
Comment (armenzg) [reply requirements-base.txt:1]: Let's say you want to update package A and run `make freeze-requirements`, would it not update everything required for package A (expected) while also updating some other packages (I would say unexpected or a side-effect).

[the remark this answers is not in the document]
Comment (armenzg) [reply requirements-base.txt:1]: Ah okay. I may have experienced that. Thanks @asottile-sentry

**Pull request closed without merging**